skyportal/handlers/api/candidate.py

```diff
@@ -399,14 +401,84 @@ def get(self, obj_id=None):
             # the LIMT/OFFSET helper function down the line once other query
             # params are set.
             order_by = [Obj.last_detected.desc().nullslast(), Obj.id]
-        if unsaved_only == "true":
-            q = q.filter(
-                Obj.id.notin_(
-                    DBSession()
-                    .query(Source.obj_id)
-                    .filter(Source.group_id.in_(group_ids))
+
+        if saved_status in [
+            "all",
+            "savedToAllSelected",
+            "savedToAnySelected",
+            "savedToAnyAccessible",
+            "notSavedToAnyAccessible",
+            "notSavedToAnySelected",
+            "notSavedToAllSelected",
+        ]:
+            if saved_status == "savedToAllSelected":
+                # Retrieve objects that have as many active saved groups that are
+                # in 'group_ids' as there are items in 'group_ids'
+                q = q.filter(
+                    Obj.id.in_(
+                        DBSession()
```

Comment:
These three lines are common to all queries, so can lift them out.

Reply:
Hmm I'm actually not sure the best way to cleanly lift them out - actually half of them use `Obj.id.notin_()` instead of `Obj.id.in_()` so not sure if lifting them out will result in cleaner code. I imagine I'd be assigning the inner queries to variables for each of these if/elif cases, but then will just end up creating another if/elif at the end for the `in_` vs `notin_` at the end anyways. @stefanv

Reply:
I think you can do `~table_a.id.in_(subquery))`?

Reply:
@stefanv common lines are now factored out